src/ai/backend/gateway/groupconfig.py

```diff
@@ -0,0 +1,252 @@
+        if len(dotfiles) == 100:
+            raise DotfileCreationFailed('Dotfile creation limit reached')
+        if not verify_dotfile_name(params['path']):
+            raise InvalidAPIParameters('dotfile path is reserved for internal operations.')
+
+        duplicate = [x for x in dotfiles if x['path'] == params['path']]
+        if len(duplicate) > 0:
+            raise DotfileAlreadyExists
+        new_dotfiles = list(dotfiles)
+        new_dotfiles.append({'path': params['path'], 'perm': params['permission'],
+                             'data': params['data']})
+        dotfile_packed = msgpack.packb(new_dotfiles)
+        if len(dotfile_packed) > MAXIMUM_DOTFILE_SIZE:
+            raise DotfileCreationFailed('No leftover space for dotfile storage')
+
+        query = (groups.update()
+                        .values(dotfiles=dotfile_packed)
+                        .where(groups.c.id == group_id))
+        await conn.execute(query)
+    return web.json_response({})
+
+
+@auth_required
+@server_status_required(READ_ALLOWED)
+@check_api_params(t.Dict({
+    tx.AliasedKey(['group', 'groupId', 'group_id']): tx.UUID | t.String,
+    t.Key('domain', default=None): t.String | t.Null,
+    t.Key('path', default=None): t.Null | t.String,
+}))
+async def list_or_get(request: web.Request, params: Any) -> web.Response:
+    resp = []
+    group_id_or_name = params['group']
+    dbpool = request.app['dbpool']
+
+    log.info('LIST_OR_GET DOTFILE (group: {0})', params['group'])
+    async with dbpool.acquire() as conn:
+        if isinstance(group_id_or_name, str):
+            if params['domain'] is None:
+                raise InvalidAPIParameters('Missing parameter \'domain\'')
+            query = (sa.select([groups.c.id])
+                        .select_from(groups)
+                        .where(groups.c.domain_name == params['domain'])
+                        .where(groups.c.name == group_id_or_name))
+            group_id = await conn.scalar(query)
+        else:
+            group_id = group_id_or_name
+        if group_id_or_name and group_id is None:
+            raise GroupNotFound
+
+        if params['path']:
+            dotfiles, _ = await query_group_dotfiles(conn, group_id)
+            if dotfiles is None:
+                raise GroupNotFound
+            for dotfile in dotfiles:
+                if dotfile['path'] == params['path']:
+                    return web.json_response(dotfile)
+            raise DotfileNotFound
+        else:
+            dotfiles, _ = await query_group_dotfiles(conn, group_id)
+            if dotfiles is None:
+                raise GroupNotFound
+            for entry in dotfiles:
+                resp.append({
+                    'path': entry['path'],
+                    'permission': entry['perm'],
+                    'data': entry['data']
+                })
+            return web.json_response(resp)
+
+
+@server_status_required(READ_ALLOWED)
+@auth_required
```

Comment:
Group dotfiles should only be updatable by superadmins, and the domain admins where the group belongs to.